app/controllers/tmdb_controller.rb

```diff
@@ -135,12 +133,11 @@ def director_search
   end
 
   def discover_search
-    form_params = %i[sort_by date genre actor actor2 company mpaa_rating year_select page]
+    form_params = %i[sort_by date year genre actor_name company mpaa_rating timeframe page]
     passed_params = params.slice(*form_params).select { |_k, v| v.present? }
     return if passed_params.blank?
 
     searchable_params = SearchParamParser.parse_movie_params(passed_params)
-    tmdb_handler_discover_search(searchable_params)
-    @params_for_view = SearchParamParser.parse_movie_params_for_display(passed_params)
+    @data = Tmdb::Client.get_advanced_movie_search_results(searchable_params)
```

Comment:
I absorbed the concept of "params for view" into the querying method. It now returns params for 2 purposes:
* a hash-friendly version for those required for pagination
* a human-friendly version for display on the view